.github/workflows/test.yml

```diff
@@ -0,0 +1,163 @@
+# SPDX-License-Identifier: Apache-2.0
+# Copyright 2022 The Foundry Visionmongers Ltd
+
+# Test building a project including OpenAssetIO and MediaCreation in
+# various submodule/cmake-find combinations.
+name: Test
+on:
+  pull_request:
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: true
+
+jobs:
+  build-openassetio:
+    name: Build OpenAssetIO
+    runs-on: ubuntu-latest
+    container:
+      image: ghcr.io/openassetio/openassetio-build
+    steps:
+      - uses: actions/checkout@v3
+      - name: Build
+        uses: ./.github/build_openassetio
+
+  build-openassetio-mediacreation:
+    name: Build OpenAssetIO-MediaCreation
+    runs-on: ubuntu-latest
+    container:
+      image: ghcr.io/openassetio/openassetio-build
+    steps:
+      - uses: actions/checkout@v3
+      - name: Build
+        uses: ./.github/build_openassetio_mediacreation
+
+  test-no-submodule:
+    name: Test (OpenAssetIO as package, MediaCreation as package)
+    runs-on: ubuntu-latest
+    container:
+      image: aswf/ci-vfxall:2022-clang14.3
+    needs: [build-openassetio, build-openassetio-mediacreation]
+    steps:
+      - uses: actions/checkout@v3
+
+      - name: Get OpenAssetIO
+        uses: actions/download-artifact@v3
+        with:
+          name: OpenAssetIO Build
+          path: ${{ github.workspace }}/openassetio-build
+
+      - name: Get OpenAssetIO-MediaCreation
+        uses: actions/download-artifact@v3
+        with:
+          name: OpenAssetIO MediaCreation Build
+          path: ${{ github.workspace }}/openassetio-mediacreation-build
+
+      - name: Configure CMake build
+        run: >
+          cmake -S .
+          -DCMAKE_PREFIX_PATH="`pwd`/openassetio-build;`pwd`/openassetio-mediacreation-build"
```

Comment:
[consider][minor] Ideally also use `${{ github.workspace }}` when setting `CMAKE_PREFIX_PATH`.

Also note again, you can set `CMAKE_PREFIX_PATH` under an `env:` (for the build step).  Doesn't really matter, just calling it out as an option in case you think its neater.

Reply:
Just noticed/reminded that `env` does have the useful property of being shown in the CI log separate from the command line and in expanded form.  A minor convenience.

Reply:
I this this when trying this, the old `github.workspace` vs `GITHUB.WORKSPACE` thing, but this time they're different whether you're in/out of a container or not. This manifested as the `download-artifact` jobs getting a different path to the `configure` jobs.

Because it's being a pain, and this is only a minor consider, i'm leaving them as is, as I don't think `pwd` is incorrect.